src/Traits/ReflectionClassLikeTrait.php

```diff
@@ -26,13 +26,18 @@
 use PhpParser\Node\Stmt\Interface_;
 use PhpParser\Node\Stmt\Trait_;
 use PhpParser\Node\Stmt\TraitUseAdaptation;
-use ReflectionObject;
-use RuntimeException;
-
-use function func_num_args;
+use ReflectionClass as BaseReflectionClass;
+use ReflectionClassConstant as BaseReflectionClassConstant;
+use ReflectionException as BaseReflectionException;
+use ReflectionExtension as BaseReflectionExtension;
+use ReflectionMethod as BaseReflectionMethod;
+use ReflectionObject as BaseReflectionObject;
+use ReflectionProperty as BaseReflectionProperty;
 
 /**
  * General class-like reflection
+ *
+ * @template T of object
  */
 trait ReflectionClassLikeTrait
```

Comment:
I dont't see any support for PHP8 attributes via AST reflection - thus, it won't work properly inside `goaop` itself, as main requirement for `parser-relfection` is to avoid loading class into memory before it will be transformed by the engine.

Is it possible to add support for PHP8 attributes via AST analysis for all reflection classes?

Reply:
Does it make sense to add more methods than the BaseReflection classes have? For example there is only `getAttributes`, but you could also add `getAttribute` and `hasAttribute`?

Reply:
I see, then it will be like that :)

Reply:
WIP https://github.com/goaop/parser-reflection/pull/122/commits/b60ced001ace42fbb28d1a6c56c63f297d941793

Reply:
Done! https://github.com/goaop/parser-reflection/pull/122/commits/4b2de97f0268f93dc766632dacbc48667d888253 with tests also https://github.com/goaop/parser-reflection/pull/122/commits/ae631c4a579285dbe4c7e48a3cd398c8ad7546c4